pkg/api/v1/types.go

```diff
@@ -1970,6 +1970,9 @@ type PodLogOptions struct {
 	// Return previous terminated container logs.
 	// Defaults to false.
 	Previous bool `json:"previous,omitempty"`
+
+	// Number of lines to show from the end of the logs
+	Tail string `json:"tail"`
```

Comment:
Must be optional, so needs omitempty.

TailLines/tailLines would be more clear.

I'd prefer this be int rather than string. We can distinguish unset from an integer value by making it a pointer to int: nil implies unset.

Are negative values allowed? Do they mean anything?

We need to define the interaction with follow.